src/utils/bind.js

```diff
@@ -5,11 +5,14 @@
  * @param {...*} arguments - Arguments to pass through.
  */
 module.exports = function bind (fn, ctx/* , arg1, arg2 */) {
-  return (function (prependedArgs) {
+  // Disable optimization in favor of not allocating memory every frame.
+  return fn.bind(ctx);
+  
+  /*return (function (prependedArgs) {
     return function bound () {
       // Concat the bound function arguments with those passed to original bind
       var args = prependedArgs.concat(Array.prototype.slice.call(arguments, 0));
```

Comment:
Did you check whether `arg1` or `arg2` are ever passed in here? If they are then it seems this replacement will lose them.

Reply:
oh you're right, this is problematic, i forgot `bind` takes an `argv`. i can allocate the `arguments` array here in this case because it's at `bind` time not call time (which is where it was before